zilencer/management/commands/populate_billing_realms.py

```diff
@@ -110,11 +127,44 @@ def handle(self, *args: Any, **options: Any) -> None:
                 tier=CustomerPlan.TIER_CLOUD_STANDARD,
                 status=CustomerPlan.FREE_TRIAL,
             ),
+            # Use `server` keyword in the unique_id to indicate that this is a profile for remote server.
+            CustomerProfile(
+                unique_id="legacy-server",
+                tier=CustomerPlan.TIER_SELF_HOSTED_LEGACY,
+            ),
+            CustomerProfile(
+                unique_id="legacy-server-upgrade-scheduled",
+                tier=CustomerPlan.TIER_SELF_HOSTED_LEGACY,
+                status=CustomerPlan.SWITCH_PLAN_TIER_AT_PLAN_END,
+                new_plan_tier=CustomerPlan.TIER_SELF_HOSTED_PLUS,
+            ),
+            CustomerProfile(
+                unique_id="business-server",
+                tier=CustomerPlan.TIER_SELF_HOSTED_BUSINESS,
+            ),
+            CustomerProfile(
+                unique_id="business-server-payment-starts-in-future",
+                tier=CustomerPlan.TIER_SELF_HOSTED_BUSINESS,
+            ),
         ]
 
-        # Create a realm for each customer profile
+        # Delete all existing remote servers
+        RemoteZulipServer.objects.all().delete()
+        flush_cache(None)
+
+        servers = []
         for customer_profile in customer_profiles:
-            populate_realm(customer_profile)
+            if "server" in customer_profile.unique_id:
```

Comment:
It would probably be better to just add an explicit `type` variable for realm/remoteserver/remoterealm than do this sort of parsing.